greenplumpython/table.py

```diff
@@ -52,6 +52,26 @@ def select(self, target_list: Iterable) -> "Table":
             parents=[self],
         )
 
+    def join(
+        self,
+        other: "Table",
+        expr: expr.Expr,
```

Comment:
1. `expr` is too general here. Better to use a more specific name such as `cond` for "condition"
2. `expr` should be optional. `CROSS JOIN` does not require any condition.

Reply:
```suggestion
        Optional[cond]: expr.Expr,
```